Automattic/wp-calypso
# Domains: Add email field to manage > "list all" page

#### Changes proposed in this Pull Request

We want to make it easier for a user to update the contact email for all of their domains. This PR adds an action to the "list all" mode of the "/domains/manage" page. This shows the email field of the contact info form and pre-fills it with the current (or pending) email address for the user's WordPress.com account.

The page should work something like this:


https://user-images.githubusercontent.com/1379730/123338864-84912180-d517-11eb-8838-cb786fc8e747.mov



Note that this version of the page only supports updating the contact email address, but there will likely be a future version that will support updating other contact fields as well.

When the contact information is saved successfully, you will see a notice indicating this:

<img width="1266" alt="Screen Shot 2021-06-29 at 7 33 51 AM" src="https://user-images.githubusercontent.com/1379730/123790824-c5d55880-d8ac-11eb-9325-6a04d6a1e3bc.png">

When no domains are selected, you will see a notice indicating this:

<img width="1266" alt="Screen Shot 2021-06-29 at 7 33 16 AM" src="https://user-images.githubusercontent.com/1379730/123790902-d7b6fb80-d8ac-11eb-9282-2268ab625bc3.png">


#### Testing instructions

Go to the page: /domains/manage?site=all&action=edit-contact-email and enter an email address. Then click on "Save contact info". Make sure that the email address is updated properly.

If you want to test the Calypso changes without actually updating your domains, hack the `/domains/<site>/whois` to return a success (or failure) response and skip the code to make the actual update.

We should also make sure that the `/domains/manage?site=all` page still works as expected
